src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -209,70 +209,81 @@ export async function initialize({
-            handlerName,
-            runtime: args.runtime,
-            workspaceUri: args.workspaceFolder.uri
-        })
-        logger.debug(`pythonCodeLensProvider.initialize: ${
-            JSON.stringify({ samProjectCodeRoot, inputTemplatePath, handlerName, manifestPath }, undefined, 2)
-            }`)
+            logger.debug(`pythonCodeLensProvider.invokeLambda: ${
+                JSON.stringify({ samProjectCodeRoot, inputTemplatePath, handlerName, manifestPath }, undefined, 2)
+                }`)
 
-        const codeDir = samProjectCodeRoot
-        const samTemplatePath: string = await executeSamBuild({
-            baseBuildDir,
-            channelLogger,
-            codeDir,
-            inputTemplatePath,
-            manifestPath,
-            samProcessInvoker: processInvoker,
+            const codeDir = samProjectCodeRoot
+            const samTemplatePath: string = await executeSamBuild({
+                baseBuildDir,
+                channelLogger,
+                codeDir,
+                inputTemplatePath,
+                manifestPath,
+                samProcessInvoker: processInvoker,
 
-        })
+            })
 
-        const debugConfig: PythonDebugConfiguration = makeDebugConfig({ debugPort, samProjectCodeRoot })
-        await invokeLambdaFunction({
-            baseBuildDir,
-            channelLogger,
-            configuration,
-            debugConfig,
-            samLocalInvokeCommand: localInvokeCommand!,
-            originalSamTemplatePath: args.samTemplate.fsPath,
-            samTemplatePath,
-            documentUri: args.document.uri,
-            originalHandlerName: args.handlerName,
-            handlerName,
-            isDebug: args.isDebug,
-            runtime: args.runtime,
-            telemetryService
-        })
-        if (args.isDebug && lambdaDebugFilePath) {
-            await unlink(lambdaDebugFilePath)
+            const debugConfig: PythonDebugConfiguration = makeDebugConfig({ debugPort, samProjectCodeRoot })
+            await invokeLambdaFunction({
+                baseBuildDir,
+                channelLogger,
+                configuration,
+                debugConfig,
+                samLocalInvokeCommand: localInvokeCommand!,
+                originalSamTemplatePath: args.samTemplate.fsPath,
+                samTemplatePath,
+                documentUri: args.document.uri,
+                originalHandlerName: args.handlerName,
+                handlerName,
+                isDebug: args.isDebug,
+                runtime: args.runtime,
+                telemetryService
+            })
+        } catch (err) {
+            const error = err as Error
+            channelLogger.error(
+                'AWS.error.during.sam.local',
+                'An error occurred trying to run SAM Application locally: {0}',
+                error
+            )
+        } finally {
+            if (lambdaDebugFilePath) {
+                await unlink(lambdaDebugFilePath)
```

Comment:
I think this throws if `lambdaDebugFilePath` does not exist. Should we wrap this line in a try/catch and swallow errors?

Reply:
I'll move the unlink call into a separate function and swallow errors to keep this section leaner